src/bindings/text-buffer-wrapper.cc

```diff
@@ -799,7 +799,14 @@ class Loader {
 
     if (progress_callback) {
       Local<Value> argv[] = {Nan::New<Number>(100), patch_wrapper};
-      auto progress_result = progress_callback->Call(2, argv, async_resource);
+      MaybeLocal<v8::Value> progress_result;
+      if (caller_async_resource) {
```

Comment:
Is there a concise way of using `caller_async_resource` or `async_resource` in C++ as we can do in JavaScript with the `||` operator?

Reply:
You *could* use a ternary expression like:

```cpp
auto resource = caller_async_resource ? caller_async_resource : async_resource;
if (resource) {
  progress_result = progress_callback->Call(2, argv, resource);
} else {
  progress_result = Nan::Call(*progress_callback, 2, argv);
}
```

Reply:
Thanks!  Ternary expression did the trick